### PR TITLE
Create add product route

### LINKED ISSUE #12: /addProduct , POST
#### Description
When seller click on add new product it has to store a new product in the database so can any user see it on the landing page

---
#### Files
- Database -> Queries -> addProductQuery.js
- Database -> Queries -> index.js
- Controllers -> addProduct.js 
- Controllers -> index.js
- Routes -> index.js -> app.post();
---

#### Tasks

- [ ] Create DB query to add a new item to the database. 
- [ ] Import the query in controllers from DB>QUERIES>index.js and validate the inputs then add to DB.
- [ ] Create POST Route in Routes>index.js

### PR DESCRIPTION
**Pull request by umarsalah**

### CODE REVIEW

**Review by ameera-abed — DISMISSED** [review]

**Review by Raghad-Mezied — DISMISSED** [review]

**Review by Ahmad-Omar — DISMISSED** [review]

File: server/controllers/addProduct.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  const {
+    name, catagory, price, description, img,
+  } = req.body;
+  const { userRole } = req;
+  if (userRole === 'seller') {
+    addProductQuery(name, catagory, price, description, img)
+      .then(() => res.status(201).json({ msg: 'Product Added Succesfully' }))
+      .catch((err) => next(err));
+  } else {
+    res.json({ msg: 'You are not a seller' });
```
Comment (LinaYahya) [inline server/controllers/addProduct.js:13]: specify status here

File: server/controllers/addUser.js [hunk 2]
```diff
@@ -2,20 +2,25 @@ const { hashPassword, signupSchema } = require('../utils/validations');
 const { addUserQuery, getUserId } = require('../database/queries');
 
 const addUser = (req, res, next) => {
-  const { error, value: { username, email, password } } = signupSchema.validate(req.body);
+  const {
+    error, value: {
+      username, email, password, userRole,
+    },
+  } = signupSchema.validate(req.body);
```
Comment (LinaYahya) [inline server/controllers/addUser.js:9]: try to validate your body asynchronously, also if you want validation to continue even If your first validator failed, add `{abortEarly: false}` as options to your validate

File: server/controllers/addUser.js [hunk 3]
```diff
@@ -2,20 +2,25 @@ const { hashPassword, signupSchema } = require('../utils/validations');
...
 
   if (error) {
     res
       .status(400)
       .json({ msg: error.details[0].message });
   } else {
     hashPassword(password)
-      .then((hashed) => addUserQuery(username, email, hashed)
+      .then((hashed) => addUserQuery(username, email, hashed, userRole)
```
Comment (LinaYahya) [inline server/controllers/addUser.js:17]: Not sure why you did those changes in addProduct branch, 
Even though did you think what might happen if you try to add a new user with an existing email?!

File: server/controllers/addUser.js [hunk 4]
```diff
@@ -2,20 +2,25 @@ const { hashPassword, signupSchema } = require('../utils/validations');
...
         .then(({ rowCount }) => {
           if (rowCount === 1) {
             getUserId(email)
               .then(({ rows }) => {
```
Comment (LinaYahya) [inline server/controllers/addUser.js:21]: avoid nested promise

File: server/controllers/addProduct.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+const { addProductQuery } = require('../database/queries');
+
+module.exports = (req, res, next) => {
+  const {
+    name, catagory, price, description, img,
+  } = req.body;
+  const { userRole } = req;
+  if (userRole === 'seller') {
+    addProductQuery(name, catagory, price, description, img)
```
Comment (LinaYahya) [inline server/controllers/addProduct.js:9]: validation goes first

**Review by LinaYahya — CHANGES_REQUESTED** [review]

**Review by Ahmad-Omar — DISMISSED** [review]

**Review by Raghad-Mezied — DISMISSED** [review]

**Review by Ahmad-Omar — APPROVED** [review]

**Review by ameera-abed — APPROVED** [review]

**Review by Raghad-Mezied — APPROVED** [review]

**Review by LinaYahya — CHANGES_REQUESTED** [review]

**Pull request closed without merging**